File: src/190.reverse-bits/190.reverse-bits.cs
/*
 * @lc app=leetcode id=190 lang=csharp
 *
 * [190] Reverse Bits
 */
/*
Accepted
600/600 cases passed (48 ms)
Your runtime beats 82.47 % of csharp submissions
Your memory usage beats 66.67 % of csharp submissions (15 MB)
*/
// @lc code=start
public class Solution {
        public uint reverseBits(uint n) {
        uint m=0;
        for(int res=0;res<32;res++){
            m=m<<1|n>>res&1;
        }
        return m;
    }
}
// @lc code=end

File: src/190.reverse-bits/190.reverse-bits.py
#
# @lc app=leetcode.cn id=190 lang=python3
#
# [190] Reverse Bits
#
# Accepted
# 600/600 cases passed (36 ms)
# Your runtime beats 93.26 % of python3 submissions
# Your memory usage beats 20.64 % of python3 submissions (13.7 MB)

# @lc code=start
class Solution:
    def reverseBits(self, n: int) -> int:
        m = 0
        res = 0
        while res < 32:
            m = m << 1 | n >> res & 1
            res += 1
        return m
        # @lc code=end
